### PR TITLE
add more bootkube recover docs

### PR DESCRIPTION
**Pull request by rphillips**
* Add some documentation on how bootkube recover works
* Remove self hosted etcd variables

### CODE REVIEW

File: Documentation/disaster-recovery.md [hunk 1]
```diff
@@ -50,7 +50,26 @@ plane.
...
 it extracts the control plane configuration from an available source and
 renders manifests in a format that `bootkube start` can use invoked to reboot
-the cluster.
+the cluster. There are two available sources to choose from: etcd or an API
+server.
+
+### What does bootkube recover do?
+
+Bootkube attempts to read the configuration from an existing backend etcd or API
+server. On success, the tool renders a modified bootstrap control plane to
```
Comment (xiang90) [inline Documentation/disaster-recovery.md]: what does `modified` mean? also can we clarify that the tool renders manifests (basically change control plane to control plane manifests).

File: pkg/recovery/recover.go [hunk 2]
```diff
@@ -80,11 +80,6 @@ type controlPlane struct {
 	daemonSets  v1beta1.DaemonSetList
 	deployments v1beta1.DeploymentList
 	secrets     v1.SecretList
-
-	// for self hosted etcd recovery if not nil
-	bootEtcd        *asset.Asset
```
Comment (xiang90) [inline pkg/recovery/recover.go]: can we separate this out to another commit? want to keep this commit cleaner.

Comment (rphillips) [reply pkg/recovery/recover.go]: split into two commits... still working on addressing the comments.

File: Documentation/disaster-recovery.md [hunk 3]
```diff
@@ -47,10 +47,31 @@ In the event of partial or total self-hosted control plane loss, `bootkube
...
+renders manifests to the local filesystem in a format that `bootkube start` can
+be invoked to reboot the cluster.
+
+There are two available sources to choose from: etcd or an API server.
+
+### What does bootkube recover do?
+
+Bootkube attempts to read the configuration from an existing backend etcd or API
+server. On success, the tool renders a modified bootstrap control plane to
+support direct injection into a kubelet.
```
Comment (colemickens) [inline Documentation/disaster-recovery.md]: Clarify what "direct injection into a kubelet" means. I assume it means, roughly, "in a format that `bootkube start` can provide to the kubelet, as static pods, to restart the control plane bootstrap process".

Comment (colemickens) [reply Documentation/disaster-recovery.md]: Or maybe more simply "in a format that `bootkube start` can handle", since the user doesn't really "directly inject" static pods, but rather just calls `bootkube start` (which is documented elsewhere).

Comment (rphillips) [reply Documentation/disaster-recovery.md]: good call.. @xiang90 @colemickens I believe the current commits address your comments.

**Comment by xiang90** [comment]
lgtm. defer to @colemickens or @ericchiang. thanks!

**Comment by rphillips** [comment]
Does the recovery control plane need to be torn down?

**Comment by ericchiang** [comment]
> Does the recovery control plane need to be torn down?

From slack. `bootkube start` already tears down the temp control plane. It'll do the same thing for the recovery control plane.
